tests/bug-detectors/path-traversal/fuzz.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint no-undef: 0 */
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+const evil_path = "../../jaz_zer/";
+const safe_path = "../../safe_path/";
+
+const fs = require("fs");
+const fsp = require("fs").promises;
+const path = require("path");
+
+/**
+ * libFuzzer tends to call the test function at least twice: once with empty data; and subsequent times with user data.
+ * If the test function generates a directory, it will fail with error "EEXIST: file already exists, mkdir '...'" on the
+ * second call. Thus, we call only once. The functions here are used in individual tests, hence, a global variable is
+ * fine.
+ */
+let called = false;
+function callOnce(fn, ...args) {
+	if (called) {
+		return;
+	}
+	fn(...args);
+	called = true;
+}
+
+// Test fs module
+module.exports.PathTraversalFsOpenEvilSync = function (data) {
+	callOnce(fs.openSync, evil_path, "r");
+};
+
+module.exports.PathTraversalFsOpenEvilAsync = async function (data) {
+	callOnce(fs.open, evil_path, "r", (err, f) => {});
```

Comment:
It could be helpful if async tests schedule the function to test for later execution, as that could uncover problems with event loop handling.

Reply:
Added one more test